### DISCUSSION TITLE
How can I show an error for an expired magic link?

### DISCUSSION THREAD

**Discussion by lumenwrites** [supabase/supabase#12939] [Questions]
When the user clicks on a magic link for the second time, they're redirected to a url that looks like this (indicating that the link has expired):
```
http://localhost:3000/#error=unauthorized_client&error_code=401&error_description=Email+link+is+invalid+or+has+expired
```
And then supabase automatically redirects them back to my home page (`/`).

So the user fails to log in, and they don't see any information explaining the issue. How can I "see" this error on the client, so I could show the user an appropriate message? 

I can't simply do something like:
```javascript
if (location.hash.includes("error")) {
  showErrorMessage()
}
```

Because the user is immediately automatically redirected to `/`.

**Comment by Chippd** [comment]
Are you sure? 
> Because the user is immediately automatically redirected to /

Surely this is some logic in your app itself. What I see in my app is the hash in the url, and I can handle that in my routing then, sending them to an /auth page with a message that explains the link is invalid or expired. 

eg: https://aihairstyles.com/#error=unauthorized_client&error_code=401&error_description=Email+link+is+invalid+or+has+expired

**Comment by fnimick** [comment]
@lumenwrites I ran into this same problem (chippd's solution works if you detect the hash client side and render an error, but it's pretty hacky). Turn out there's a PR in the works to fix. https://github.com/supabase/gotrue/pull/1129

**Comment by Nicolasvegam** [comment]
```
  useEffect(() => {
    if (router.asPath ==
      "/#error=unauthorized_client&error_code=401&error_description=Email+link+is+invalid+or+has+expired") {
      setLinkError('Write something...');
    }
  }, [router.asPath]);
```

this was my workaround, don't like tho

**Comment by catlan** [comment]
```
    if (document.location.hash.startsWith('#')) {
      const params = new URLSearchParams(document.location.hash.substring(1));
      setError(params.get('error'));
      setErrorCode(params.get('error_code'));
      setErrorDescription(params.get('error_description'));
    }
```
